### ISSUE TITLE
Transaction is not working

### ISSUE THREAD

**Issue by alickmail** [sebelga/gstore-node#144]
I am not able to use transaction for model update, the `transaction.commit().then()` is fired, but the data on google console did not updated. 

I have tried the Model.update function without using transaction, the data on google console is updated successfully

 i am following the example from gitbook:
[Link](https://sebloix.gitbook.io/gstore-node/model/methods/update-method)

# Here is update model code:
```javascript
function transferCoins(fromUser, toUser, coins) {
  return new Promise((resolve, reject) => {
    let fromUserObj = fromUser.plain();
    let toUserObj = toUser.plain();

    let transaction = gstore.transaction();
    transaction.run().then(() => {
      User.update(fromUserObj.id, {
        name : 'asdsadfssssf',
        coins: fromUserObj.coins - 1000
      }, null, null, transaction);

      transaction.commit().then((x) => {
        resolve();
      }).catch(err => {
        console.log({err});
      });

      // User.update(toUserObj.id, toUserObj.coins + 1000, null, null, transaction);
    }).catch(err => {
      transaction.rollback();
      reject(err);
    });
  });
}
```

# Here is the model file
```javascript
const gstore = require('gstore-node')();
const Schema = gstore.Schema;
const Joi = require('joi');

const userSchema = new Schema({
  name        : {joi: Joi.string().required()},
  lastname    : {joi: Joi.string()},
  password    : {joi: Joi.string()},
  coins       : {joi: Joi.number().integer().min(0)},
  email       : {joi: Joi.string().email()},
  createdAt   : {joi: Joi.date()},
  access_token: {joi: [Joi.string(), Joi.number()]},
  birthyear   : {joi: Joi.number().integer().min(1900).max(2013)},
}, {joi: true});


const User = gstore.model('User', userSchema);

module.exports = User;

```

Please help, thanks!

**Comment by sebelga** [comment]
Hello,
Thanks for reporting, your code looks ok so I will try to reproduce and let you know.
cheers

**Comment by sebelga** [comment]
Hi @alickmail 

I just released `v5.0.0` that should fix this bug. There was indeed a bug in the `Model.update()` method when using transaction.

Once you update the gstore version (make sure you update how you instantiate gstore, there is a breaking change), you will need to make a small change in your code:

```js
...
function transferCoins(fromUser, toUser, amount) {
    const transaction = gstore.transaction();
    return transaction.run()
        .then(async () => {
            await User.update(fromUser.entityKey.id, {
                coins: fromUser.coins - amount,
            }, null, null, transaction);

            await User.update(toUser.entityKey.id, {
                coins: toUser.coins + amount,
            }, null, null, transaction);

            return transaction.commit();
        }).catch((err) => {
            transaction.rollback();
            throw err;
        });
}


// ----------------------------
// Or without "async" and "waiting" if you already have fetched the entity data previously

function transferCoins(fromUser, toUser, amount) {
    const transaction = gstore.transaction();
    return transaction.run()
        .then(() => {
            const fromUserUpdated = new User({
                ...fromUser.entityData,
                coins: fromUser.coins - amount},
                fromUser.entityKey.id
            );

            const toUserUpdated = new User({
                ...toUser.entityData,
                coins: toUser.coins + amount},
                toUser.entityKey.id
            );

            fromUserUpdated.save(transaction);
            toUserUpdated.save(transaction);

            return transaction.commit();
        }).catch((err) => {
            transaction.rollback();
            throw err;
        });
}
```

**Comment by sebelga** [comment]
You use case made me think that it'd be useful to add a sort of `updater()` or `increment` functionality to dyncamically update a value.

Because you might run in a race condition if 2 requests fetch the `fromUser` entity, have the same "coins" and then try to update the same user... you might want to think in a "lock" version system when fetching the user so when a second request try to update the user and the lock version mismatch you cancel the request.
